packages/dd-trace/src/appsec/waf/waf_context_wrapper.js

```diff
@@ -32,13 +33,20 @@ class WAFContextWrapper {
 
       const end = process.hrtime.bigint()
 
+      const ruleTriggered = !!result.data && result.data !== '[]'
+      const requestBlocked = result.actions?.includes('block')
```

Comment:
Bad naming, the request hasn't been blocked at this point, it should be something like `shouldBlock`. Conveying that the WAF returned a blocking action, but that it hasn't been blocked yet

Reply:
Just realized it's the name of the telemetry metric itself that's badly named :/

Reply:
`request_blocked` is the metric tag's name but at this point may not be very appropriate. 
Do you like `blockTriggered`?